Formula/lib/librime.rb

```diff
@@ -0,0 +1,74 @@
+class Librime < Formula
+  desc "Rime Input Method Engine"
+  homepage "https://rime.im"
+  url "https://github.com/rime/librime.git",
+      tag:      "1.12.0",
+      revision: "c7ab6390c143a11d670f3add41218111edb883c9"
+  license "BSD-3-Clause"
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "icu4c@76" => :build
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => :build
+  depends_on "python@3.13" => :build
+
+  depends_on "lua"
+
+  on_linux do
+    depends_on "libunwind"
+  end
+
+  resource "lua" do
+    url "https://github.com/hchunhui/librime-lua.git",
+        revision: "b210d0cfbd2a3cc6edd4709dd0a92c479bfca10b"
+  end
+
+  resource "octagram" do
+    url "https://github.com/lotem/librime-octagram.git",
+        revision: "dfcc15115788c828d9dd7b4bff68067d3ce2ffb8"
+  end
+
+  resource "predict" do
+    url "https://github.com/rime/librime-predict.git",
+        revision: "920bd41ebf6f9bf6855d14fbe80212e54e749791"
+  end
+
+  def install
+    resources.each do |r|
+      r.stage buildpath/"plugins"/r.name
+    end
+
+    ENV.append_to_cflags "-fPIC" if OS.linux?
+
+    # TODO: Dependents should moved to `depends_on` block, but it is not work now
+    system "make", "deps"
```

Comment:
We shouldn't be vendoring these dependencies: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
Yeap, I tried to import those libraries through the homebrew packages but it does not work and I think upstream should change the way to build their package. If this is not allowed then this is the end.

Reply:
@Freed-Wu , @tshu-w 
As of the policy, I cannot proceed to the next step. You may copy and paste this formula definition to your custom tap and use it. In order to register `librime` here, `deps` should be replaced by external libraries not built by your own.

Reply:
@rime maybe upstream know how to fix it?

Reply:
What is the error when we use Homebrew-provided dependencies?

Reply:
Build this library and add it to the emacs, run rime and check it for success or not. When I build this library, the size will vary depends on how I build and I suspect that the `librime` include some of static library itself. So, When I use homebrew built packages as `depends_on`, library size is a bit smaller than usual and so I use `make deps`.

I will try to follow the way `archlinux` used, when I have a time for this. Let's start a new attepmt!